src/actions/guides/frontend/internationalization.cr

```diff
@@ -0,0 +1,341 @@
+      #
+      # Add a `page_title` method to pages to override it. You can also remove
+      # This method so every page is required to have its own page title.
+      def page_title
+        I18n.t("default.page_name", @translator.lang)
+      end
+
+      def render
+        html_doctype
+
+        html lang: @translator.lang do
+          mount Shared::LayoutHead.new(page_title: page_title, context: @context)
+
+          body do
+            mount Shared::FlashMessages.new(@context.flash)
+            content
+          end
+        end
+      end
+    end
+    ```
+
+    ```
+    # src/pages/password_reset_requests/new_page.cr
+    class PasswordResetRequests::NewPage < AuthLayout
+      needs operation : RequestPasswordReset
+
+      def content
+        h1 I18n.t("auth.reset_pwd_request", @translator)
+        render_form(@operation)
+      end
+
+      private def render_form(op)
+        form_for PasswordResetRequests::Create do
+          mount Shared::Field.new(op.email), &.email_input
+          submit I18n.t("auth.reset_pwd_request", @translator), flow_id: "request-password-reset-button"
+        end
+      end
+    end
+    ```
+
+    ```
+    # src/pages/password_resets/new_page.cr
+    class PasswordResets::NewPage < AuthLayout
+      needs operation : ResetPassword
+      needs user_id : Int64
+
+      def content
+        h1 I18n.t("auth.update_pwd", @translator)
+        render_password_reset_form(@operation)
+      end
+
+      private def render_password_reset_form(op)
+        form_for PasswordResets::Create.with(@user_id) do
+          mount Shared::Field.new(op.password), &.password_input(autofocus: "true")
+          mount Shared::Field.new(op.password_confirmation), &.password_input
+
+          submit I18n.t("auth.update_pwd", @translator), flow_id: "update-password-button"
+        end
+      end
+    end
+    ```
+
+    ```
+    # src/pages/sign_ins/new_page.cr
+    class SignIns::NewPage < AuthLayout
+      needs operation : SignInUser
+
+      def content
+        h1 I18n.t("auth.sign_in", @translator)
+        render_sign_in_form(@operation)
+      end
+
+      private def render_sign_in_form(op)
+        form_for SignIns::Create do
+          sign_in_fields(op)
+          submit I18n.t("auth.sign_in", @translator), flow_id: "sign-in-button"
+        end
+        link I18n.t("auth.reset_pwd_request", @translator), to: PasswordResetRequests::New
+        text " | "
+        link I18n.t("auth.sign_up", @translator), to: SignUps::New
+      end
+
+      private def sign_in_fields(op)
+        mount Shared::Field.new(op.email), &.email_input(autofocus: "true")
+        mount Shared::Field.new(op.password), &.password_input
+      end
+    end
+    ```
+
+    ```
+    # src/pages/sign_ups/new_page.cr
+    class SignUps::NewPage < AuthLayout
+      needs operation : SignUpUser
+
+      def content
+        h1 I18n.t("auth.sign_up", @translator)
+        render_sign_up_form(@operation)
```

Comment:
I think these could be clarified by removing most of the code that doesn't need to be changed. For example: 

```crystal
    class SignUps::NewPage < AuthLayout
      def content
        h1 I18n.t("auth.sign_up", @translator)
        # ...
      end

      private def render_sign_up_form(op)
        form_for SignUps::Create do
          # ...
          submit I18n.t("auth.sign_up", @translator), flow_id: "sign-up-button"
        end
        link I18n.t("auth.sign_in", @translator), to: SignIns::New
      end

      # ...
    end
```

Reply:
OK - I was thinking then the person could copy and paste.  But this clarifies what to do better.  I have tried to cleanup - but we can make further cleanups too.